tests/subsys/net/openthread/rpc/server/src/message_suite.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <mock_nrf_rpc_transport.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_message.h>
+#include <test_rpc_env.h>
+
+#include <zephyr/fff.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#include <openthread/message.h>
+#include "common_fakes.h"
+#include <assert.h>
+#include <stddef.h>
+
+static void nrf_rpc_err_handler(const struct nrf_rpc_err_report *report)
+{
+	zassert_ok(report->code);
+}
+
+#define FOREACH_FAKE(f)                                                                            \
+	f(otMessageGetLength);                                                                     \
+	f(otMessageGetOffset);                                                                     \
+	f(otMessageRead);                                                                          \
+	f(otMessageFree);                                                                          \
+	f(otMessageAppend);                                                                        \
+	f(otUdpNewMessage);
+
+static void tc_setup(void *f)
+{
+	mock_nrf_rpc_tr_expect_add(RPC_INIT_REQ, RPC_INIT_RSP);
+	zassert_ok(nrf_rpc_init(nrf_rpc_err_handler));
+	mock_nrf_rpc_tr_expect_reset();
+
+	FOREACH_FAKE(RESET_FAKE);
+	FFF_RESET_HISTORY();
+}
+
+ZTEST(ot_rpc_message, test_otUdpNewMessage_failing)
+{
+	otUdpNewMessage_fake.return_val = NULL;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(0), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_NULL));
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(0), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_TRUE, CBOR_UINT8(40)));
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(otUdpNewMessage_fake.call_count, 2);
+}
+
+ZTEST(ot_rpc_message, test_otUdpNewMessage_free_working)
+{
+	char buf[16];
+
+	otUdpNewMessage_fake.return_val = (otMessage *)buf;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(1), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_NULL));
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(2), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_TRUE, CBOR_UINT8(40)));
+
+	zassert_equal(otUdpNewMessage_fake.call_count, 2);
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_MESSAGE_FREE, 1));
```

Comment:
Would be great to verify that `otMessageFree` has received the right pointer.

Reply:
fixed.